File: Sort排序算法/bubble-sort冒泡排序/bubble-sort.js
// bubble sort
// 从小到大排序

// 方法一
function bubbleSort (nums) {
    for (let i = 0; i < nums.length; i ++) {
        for (let j = i + 1; j < nums.length; j ++) {
            if (nums[i] > nums[j]) {
                let tmp = nums[i]
                nums[i] = nums[j]
                nums[j] = tmp
            }
        }
    }
    return nums
}

console.log(bubbleSort([1, 6, 2, 7, 4]))

File: Sort排序算法/bubble-sort冒泡排序/bubble-sort.py
# bubble sort
# 从小到大排序

# 方法一
def bubble_sort(nums):
    total = len(nums)
    for i in range(0, total):
        for j in range(i+1, total):
            if nums[i] > nums[j]:
                tmp = nums[i]
                nums[i] = nums[j]
                nums[j] = tmp
    return nums

print(bubble_sort([1, 6, 2, 7, 4]))








